brave/brave-core
# refactor(wallet): async/await in WalletUserAssetManager

Resolves https://github.com/brave/brave-browser/issues/39558
Refactor `WalletUserAssetManager` to use async/await instead of completion block. This refactoring is necessary to help implement https://github.com/brave/brave-browser/issues/39098



## Submitter Checklist:

- [x] There is a [ticket](ttps://github.com/brave/brave-browser/issues/39558) for my issue
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
Wallet should work as before. No crash should happen after this refactoring. 
Suggested test scenarios:
1. NFT hide/unhide/delete
2. add/remove custom assets
3. wallet restore with auto-discovery assets
4. dapp add suggested assets
5. add/remove custom networks
6. edit asset visibility
